akka-projection-cassandra/src/main/scala/akka/projection/cassandra/internal/CassandraOffsetStore.scala

```diff
@@ -29,32 +29,45 @@ import akka.stream.alpakka.cassandra.scaladsl.CassandraSession
   // FIXME make keyspace and table names configurable
   val keyspace = "akka_projection"
   val table = "offset_store"
+  val cassandraPartitions = 5
+
+  private[cassandra] val partitionClause = (0 until cassandraPartitions).mkString(",")
 
   def this(session: CassandraSession)(implicit ec: ExecutionContext) =
     this(session, Clock.systemUTC())
 
   def readOffset[Offset](projectionId: ProjectionId): Future[Option[Offset]] = {
     session
-      .selectOne(s"SELECT offset, manifest FROM $keyspace.$table WHERE projection_id = ?", projectionId.id)
-      .map { maybeRow =>
-        maybeRow.map(row => fromStorageRepresentation[Offset](row.getString("offset"), row.getString("manifest")))
+      .selectAll(
+        s"SELECT projection_key, offset, manifest FROM $keyspace.$table WHERE projection_name = ? AND partition IN ($partitionClause)",
+        projectionId.name)
+      .map { maybeRows =>
+        maybeRows
+          .find(_.getString("projection_key") == projectionId.key)
+          .map(row => fromStorageRepresentation[Offset](row.getString("offset"), row.getString("manifest")))
       }
   }
 
-  def saveOffset[Offset](projectionId: ProjectionId, offset: Offset): Future[Done] =
+  def saveOffset[Offset](projectionId: ProjectionId, offset: Offset): Future[Done] = {
+    // a partition is calculated to ensure some distribution of projection rows across cassandra nodes, but at the
+    // same time let us query all rows for a single projection_name easily
+    val partition = Integer.valueOf(Math.abs(projectionId.key.hashCode()) % cassandraPartitions)
```

Comment:
`abs` should be around all of it, because `math.abs(Int.MinValue)` is negative

```
Integer.valueOf(math.abs(projectionId.key.hashCode % cassandraPartitions))
```

Reply:
Interesting.  I wasn't aware that `Math.abs(Int.MinValue)` was negative.  Would the result of a mod operation where the LHS is positive ever be negative?  I'll update it anyway.